src/core_plugins/kibana/public/dashboard/panel/panel_header/panel_actions/get_inspector_panel_action.tsx

```diff
@@ -42,27 +46,30 @@ export function getInspectorPanelAction({ closeContextMenu, panelTitle }) {
     },
     {
       icon: <EuiIcon type="inspect" />,
+      isVisible: ({ embeddable }) =>
+        embeddable && Inspector.isAvailable(embeddable.getInspectorAdapters()),
       onClick: ({ embeddable }) => {
         closeContextMenu();
-        const session = Inspector.open(embeddable.getInspectorAdapters(), {
-          title: panelTitle,
-        });
-        // Overwrite the embeddables.destroy() function to close the inspector
-        // before calling the original destroy method
-        const originalDestroy = embeddable.destroy;
-        embeddable.destroy = () => {
-          session.close();
-          if (originalDestroy) {
-            originalDestroy.call(embeddable);
-          }
-        };
-        // In case the inspector gets closed (otherwise), restore the original destroy function
-        session.on('closed', () => {
-          embeddable.destroy = originalDestroy;
-        });
+        const adapters = embeddable.getInspectorAdapters();
+        if (adapters) {
```

Comment:
Not important, but I'd prefer an early exit, so I don't have to keep track of so many closing `}`. So,

```
if (adapters) {
  return;
}
```